### ISSUE TITLE
SDL menu usability issues because menu size and scaling

### ISSUE THREAD

**Issue by LukasThyWalls** [joncampbell123/dosbox-x#5249]
### Describe the bug

Hello.

I noted issues with the SDL built-in menu to reach certain menus and items when the relative size of the menu and the DOSBox-X window size are very close. It's the same issue noted here #2877 in reality, but It could be trigger in bigger window sizes. 

[As explained here](https://github.com/joncampbell123/dosbox-x/issues/877#issuecomment-418227155), the menu first doubles the size to maintain readability in bigger resolution size, and that's fine. The jump happens in 1280 pixels horizontal (1280x800).

When the menu is double, and for example, the DOSBox-X window size is 1280x960 (bigger rounded 4:3 resolution to fit in a 1080p display) some items are inaccessible because the menu need to be rendered inside the window itself, as opposite when DOSBox-X on Windows is compiled to use its menus (There I have a 1280x960 DOSBox-X window and there is no issues with the menus).

To show the difference, there is the DOSBox-X with window size 1279x960:

Main Window at 1279x960 (tiny menu):
![DOSBox-X_1279x960_Main](https://github.com/user-attachments/assets/46f248f2-bcdd-454f-b4a3-b37f0c45da1d)

Config tool at 1279x960 (tiny window, bit still visible):
![DOSBox-X_1279x960_ConfigTool](https://github.com/user-attachments/assets/2f40d9c0-fdc2-4800-ac73-563b8c6526d1)

Drive menu at 1279x960 (tiny menu, but all options are reachable):
![DOSBox-X_1279x960_DriveMenu](https://github.com/user-attachments/assets/35b93097-2922-4568-b6c0-ceff5fee3137)

One pixel more, 1280x960, the menu doubles the size:

Main Window at 1280x960 (big menu):
![DOSBox-X_1280x960_Main](https://github.com/user-attachments/assets/cdab1ba2-7319-43d1-843e-48d623c8b000)

Config tool at 1280x960 (big window, good visibility):
![DOSBox-X_1280x960_ConfigTool](https://github.com/user-attachments/assets/8df3bed8-7f60-474b-aed8-c01e5ad05ef2)

Drive menu at 1280x960 (big menu, but all options are **NOT** reachable. You can't select options from other drives than _A_!):
![DOSBox-X_1280x960_DriveMenu](https://github.com/user-attachments/assets/8ab2a2fc-9fa3-4af2-8449-fa72b51719f9)

In video, to see it in action:
[DOSBox-X_1280x960_DriveHelpMenuIssue.webm](https://github.com/user-attachments/assets/54f596b1-178f-4266-abc7-a2467feb948e)

In #2877, the issue was using the _original_ window resolution (640x480) but without doubling the menu size, the unreachable menus are the same. 

### Steps to reproduce the behaviour

1. Use DOSBox-X with built-in SDL menus (Flatpak Linux uses them).
2. Set window resolution to 1280x960 (SDL section, `windowresolution` in _dosbox-x.conf_).
3. Use It in window mode.
4. Try to reach all letters in the Drive menu or other ones with sub-menus. 

### Expected behavior

All items should be reachable.

### What operating system(s) this bug have occurred on?

KUbuntu 24.10, KDE Plasma 6.1.5, 1920x1080 (125% Scale)

### What version(s) of DOSBox-X have this bug?

DOSBox-X 2024.10.01 (Linux SDL2 64 bit). Flatpak version. Language es_ES.

### Used configuration

```ini
Only setting changed from default is:

[sdl]
(...)
windowresolution  = 1280x960
(...)
```

### Additional information

I know that have a bigger menu size is needed on bigger resolutions, but the jump at 1280 is a big one for using at that resolution or a little up. This was more or less talked here #877.

The problem here is not from the resolution itself, is how those menus work, as they overlap over the previous ones.

Some short or long solutions I could imagine could be:
- Have a between 1x - 2x menu size to use at a lower resolution (1.5x, but using the same menu with that multiplier for sure it gonna be look ugly).
- Redesign the menus, or at least, how they overlap between them, or how the sub-menus appear (click to appear sub-menu setting?).
- Set an option to force the menu scaling, setting _auto_ as default like it works, setting 1, 2, 3... as the multiply factor.

Thanks in advance!

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by maron2000** [comment]
Can you check out the nightly build after PR #5263 is merged?

**Comment by LukasThyWalls** [comment]
> Can you check out the nightly build after PR #5263 is merged?

First, I also realised something, with DOSBox-X as default in English there is no issues, but with other languages (In my case, Spanish `[dosbox] language = es_ES.lng`), the issue is triggered because the menu items had different character length, an issue you don't have, for example, in Windows with DOSBox-X OS' menus. 

Anyway, testing your version, It's rearranged the _DOS commands_ item in _Help_, and It looks like a little gap has been added to the menu to have a little position to point with the mouse:

[DOSBox-X_1280x960_DriveHelpMenuIssue_Fix1.webm](https://github.com/user-attachments/assets/b3d9edd0-f778-4a01-bcf5-e2ccfd42a1e7)

It works... mostly yes, although that little gap is a bit tricky, and I don't know if in other language some item in those menus that has more letters it could work. As a quick fix... yeah, but I don't know if its more feasible to put an option to override the menu scaling, because although the menu will look tiny It will work much better in my case.

Thanks!
